lib/ansible/modules/windows/win_pester.ps1

```diff
@@ -0,0 +1,95 @@
+#!powershell
+# This file is part of Ansible
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+$ErrorActionPreference = 'Stop'
+
+$params = Parse-Args -arguments $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+$diff_mode = Get-AnsibleParam -obj $params -name "_ansible_diff" -type "bool" -default $false
+
+# Modules parameters
+
+$src = Get-AnsibleParam -obj $params -name "src" -type "string" -failifempty $true
+$version = Get-AnsibleParam -obj $params -name "version" -type "string" -failifempty $false
+
+$result = @{
+    changed = $false
+}
+
+if ($diff_mode) {
+    $result.diff = @{}
+}
+
+# CODE
+# Test if parameter $version is valid
+Try {
+    $version = [version]$version
+}
+Catch {
+    Fail-Json -obj $result -message "$version is not a valid version format"
+}
+
+# Import Pester module if available
+$Module = 'Pester'
+
+If (-not (Get-Module -Name $Module -ErrorAction SilentlyContinue)) {
+    If (Get-Module -Name $Module -ListAvailable -ErrorAction SilentlyContinue) {
+        Import-Module $Module
+    } else {
+        Fail-Json -obj $result -message "Cannot find module: $Module"
```

Comment:
Just a little thing, but it would be nice to prompt the user to check if pester is installed and suggest they install it via win_psmodule or win_chocolatey as appropriate.

Reply:
What's the best way to prompt informations to users ?

Reply:
Sorry I could have been clearer.  I just mean it would be nice to add more information to the -message string to suggest to the user how they could fix the problem where the pester module cannot be loaded by Get-Module.  Something like
``Fail-Json -obj $result -message "Cannot find module: $Module. Check if pester is installed, and if it is not, install using win_psmodule or win_chocolatey."
``

Reply:
Thanks @jhawkesworth I've modified the code with your recommendations .